sermant-plugins/sermant-flowcontrol/flowcontrol-common/src/main/java/io/sermant/flowcontrol/common/handler/retry/policy/RetryOnUntriedPolicy.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright (C) 2024-2024 Sermant Authors. All rights reserved.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+
+package io.sermant.flowcontrol.common.handler.retry.policy;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Retry on the Untried instance, regardless of thread safety, only on thread variables
+ *
+ * @author zhp
+ * @since 2024-11-28
+ */
+public class RetryOnUntriedPolicy implements RetryPolicy {
+    private final int attempts;
+
+    private final List<Object> retryInstance;
+
+    private int hasTriedCount;
+
+    private boolean isRetry;
+
+    private boolean isFirstMark = true;
+
+    /**
+     * retry constructor
+     *
+     * @param attempts Maximum Retry Count
+     */
+    public RetryOnUntriedPolicy(int attempts) {
+        this.attempts = attempts;
+        retryInstance = new ArrayList<>();
+    }
+
+    @Override
+    public boolean needRetry() {
```

Comment:
isRetryThresholdReached

Reply:
The current class only implements the interface defined by the original interface.